readme.md

```diff
@@ -1,252 +1,262 @@
-
-1. From the "Overview" tab on your MongoDB cluster, click the "Connect" button.
-
-1. Click "Connect from Mongo Shell"
-
-1. Follow the instructions for installing MongoDB locally on your system
-
-1. Copy the command in the final step of the instructions and paste that into your command line
-
-1. Enter your password as defined in your `nodemon.json` file
-
-1. Once connected, try running the `show dbs` command
-
-1. Connect to your database using the `use <database-name>` command
-
-1. Run the following:
-  ```
-  db.guests.createIndex({ "username": 1}, { unique: true })
-  ```
-
+# Authentication
+
+By the end of this lesson, you should be able to authorize your routes with JWTs and securely store passwords.
+
+## Core Learning Objective
+
+*	Implement a basic authentication setup using JSON web tokens
+
+## Sub-Objectives
+
+* Describe the authentication process
+* Store passwords with bcrypt
+* Create signin and login routes that return JWTs
+* Describe the difference between authentication and authorizations
+* Authorize certain routes and information
+
+## Installation
+
+1. Fork & clone
+
+1. `cp nodemon.sample.json nodemon.json`
+
+1. Create a new Cluster on [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) titled something like `general-purpose` or reuse one you already have.
+
+1. Update the `MONGO_DB_CONNECTION` in your `nodemon.json` file with the full connection string. Make sure you include the password you set up for the database and change the name of the database from `test` to something lke `exclusive_party_dev`.
+
+1. `npm install`
+
+1. `npm run reset-db`
+
+1. `npm run dev`
+
+Once installation is working, take a look at the existing code to make sure you understand what is happening. Then, try making requests to the API.
+
+### Instructions & Guiding Questions
+
+- [ ] Take a look at the `db/seeds.js` file.
+
+* **Question:** Describe what this code is doing and what its purpose is.
+
+* **Your Answer:** 
+
+---Whenever `npm run reset-db` is ran, the function `reset()` within `db/seed.js` file is executed. This function first calls the `deleteMany()` method of the `Party` collection, which eliminates any existing documents from the collection. Then the `create()` method is called and two initial documents are created. This is a useful method to reset/reformat to a "default" or "foundation" instance of your collection.
+
+- [ ] Imagine that as a user, you enter your username and password into a site in order to signup.
+
+* **Question:** What happens next?
+
+* **Your Answer:**
+
+--- Assuming that various validations are performed against the username and password. For user name, system may check for format, type, etc. or for whether or not the user name already exists. For password, may check for format, type, length requirements, etc. If all validations are successful, a user/password record/collection is created.
+

```

Comment:
Good! also checks if similar records already exist and uses encryption to save to db